lib/migrator.js

```diff
@@ -1,90 +1,88 @@
 'use strict';
 
-const FeedParser = require('feedparser');
-const async = require('async');
 const TurndownService = require('turndown');
-const request = require('request');
-const url = require('url');
-const fs = require('fs');
-
-exports.registerMigrator = function(hexo) {
-  hexo.extend.migrator.register('rss', (args, callback) => {
-    const source = args._.shift();
-    const tomd = new TurndownService();
-
+const got = require('got');
+const { parse } = require('url');
+const { readFile } = require('hexo-fs');
+const parseFeed = require('./feed');
+
+module.exports = async function(args) {
+  const source = args._.shift();
+  const tomd = new TurndownService();
+  const { log } = this;
+  const Post = this.post;
+  let untitledPostCounter = 0;
+  let errNum = 0;
+  let input, feed;
+  const posts = [];
+
+  try {
     if (!source) {
       const help = [
         'Usage: hexo migrate rss <source> [--alias]',
         '',
         'For more help, you can check the docs: http://hexo.io/docs/migration.html'
       ];
 
-      console.log(help.join('\n'));
-      return callback();
+      throw help.join('\n');
     }
 
-    const log = hexo.log;
-    const post = hexo.post;
-    let untitledPostCounter = 0;
-    let stream;
-
     if (/^http(s)?:\/\//i.test(source)) {
-      stream = request(source);
+      input = await got(source, { resolveBodyOnly: true, retry: 0 });
```

Comment:
`retry` is disabled to pass the unit test (in local machine with slow connection).
The `got` options can be exposed to config in a separate PR.